modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition.py

```diff
@@ -137,19 +133,17 @@ def drain_call_queue(self):
         if len(self.call_queue) == 0:
             return
         call_queue = self.call_queue
-        client = default_client()
         if len(call_queue) > 1:
-            future = client.submit(
-                apply_list_of_funcs, call_queue, self.future, pure=False
+            futures = DaskWrapper.deploy(
+                apply_list_of_funcs, 2, call_queue, self.future, pure=False
             )
         else:
             # We handle `len(call_queue) == 1` in a different way because
             # this improves performance a bit.
             func, args, kwargs = call_queue[0]
-            future = client.submit(apply_func, self.future, func, *args, **kwargs)
-        futures = [
-            client.submit(lambda l, i: l[i], future, i, pure=False) for i in range(2)
-        ]
+            futures = DaskWrapper.deploy(
+                apply_func, 2, self.future, func, *args, **kwargs
+            )
```

Comment:
shouldn't we pass `pure=False` here as well?

Reply:
should